packages/docs-app/src/examples/core-examples/toastCreateAsyncExample.tsx

```diff
@@ -88,12 +88,13 @@ async function showMessageFromNewToaster() {
 /**
  * @param containerElement The container argument passed to OverlayToaster.create/OverlayToaster.createAsync
  */
-function unmountReact16Toaster(containerElement: HTMLElement) {
+function unmountToaster(containerElement: HTMLElement) {
     const toasterRenderRoot = containerElement.firstElementChild;
     if (toasterRenderRoot == null) {
         throw new Error("No elements were found under Toaster container.");
     }
-    ReactDOM.unmountComponentAtNode(toasterRenderRoot);
+    const root = ReactDOM.createRoot(toasterRenderRoot);
```

Comment:
I believe this calls `ReactDOM.createRoot` on a DOM node that previously had a react root created on it, which leaks memory. The old root never gets unmounted.

The `OverlayToaster.createAsync` call above doesn't seem to override the `domRenderer` field. I think the React root that was previously created on `toasterRenderRoot` is actually a React 17 legacy root. Without changes to how `createAsync` is called above, I think `ReactDOM.unmountComponentAtNode` is most correct here?

To fully remove `unmountComponentAtNode`, there would need to be a change to the `createAsync` call to create a React root, store a reference to it in the example here, and then call unmount in the cleanup effect.

Reply:
Ah, yup, good catch! I'll split this out, ignore the deprecated ReactDOM lint errors, and track this in #7166

Reply:
595f50be28e84d6845240a90416855faef095c65 / 89ce662ec2694accab04244454a9225bc0b527e5